examples/evm_bots.py

```diff
@@ -443,220 +490,300 @@ def create_agent(
     return agent
 
 
-def get_agents(
-    config: simulators.Config,
-    hyperdrive_contract: ContractInstance,
-    base_contract: ContractInstance,
-) -> dict[str, elfpy_agent.Agent]:
-    """Get python agents & corresponding on-chain accounts.
+def set_up_agents(experiment_config: simulators.Config):
+    """Set up python agents & corresponding on-chain accounts.
 
-    Returns
-    -------
-    _sim_agents : dict[str, Agent]
-        Dictionary of agents.
-    dev_accounts : list[KeyfileAccount]
-        List of dev accounts.
+    Parameters
+    ----------
+    experiment_config : simulators.Config
```

Comment:
As a rule of thumb we should be weary about functions that modify things in-place. It can cause confusion, since variables are modified "behind the scenes". Certain operations make sense to do so in-place, but they're typically member functions of classes. e.g. `my_array.sort()` might not return something, and instead just sort the array in-place.